lib/kafka/consumer.rb

```diff
@@ -277,7 +278,6 @@ def each_batch(min_bytes: 1, max_wait_time: 1, automatically_mark_as_processed:
 
           @heartbeat.send_if_necessary
 
-          @running = false if exit_if && exit_if.call
```

Comment:
Hum, since this is inside the `batches.each` which I moved outside, I wonder if I should `undelete` it?
The tests do pass with this line removed. Seems like there isn't anything that would loop or take too much time inside the `batches.each` loop. Thoughts?

Reply:
Agree that this should probably stay (although I would add a `break` and move it to before the `heartbeat.send_if_necessary` since I've seen the heartbeat take forever too.